crypto/ube/fork_detect.c

```diff
@@ -44,25 +62,29 @@ static struct CRYPTO_STATIC_MUTEX fork_detect_lock = CRYPTO_STATIC_MUTEX_INIT;
 // assume that it has exclusive access to it.
 static volatile char *fork_detect_addr = NULL;
 static uint64_t fork_generation = 0;
-static int ignore_madv_wipeonfork = 0;
 
-static int init_fork_detect_madv_wipeonfork(void **addr_out) {
+static int ignore_all_fork_detection(void) {
+  if (ignore_wipeonfork == 1 &&
+      ignore_inheritzero == 1) {
+    return 1;
+  }
+  return 0;
+}
 
-  void *addr = MAP_FAILED;
-  long page_size = 0;
+#if defined(OPENSSL_LINUX)
 
-  GUARD_PTR(addr_out);
+#include <sys/mman.h>
+#include <unistd.h>
 
-  page_size = sysconf(_SC_PAGESIZE);
-  if (page_size <= 0) {
-    return 0;
-  }
+#if defined(MADV_WIPEONFORK)
+OPENSSL_STATIC_ASSERT(MADV_WIPEONFORK == 18, MADV_WIPEONFORK_is_not_18)
+#else
+#define MADV_WIPEONFORK 18
+#endif
```

Comment:
* If `MADV_WIPEONFORK` were not already defined in the headers provided by the system, why do we think it's safe to define the value here?
* Since it's a syscall parameter, the value of `MADV_WIPEONFORK` should be stable for a given target, but how do we know that the value is 18 across all Linux targets?
    * Possibly related: `MADV_WIPEONFORK` is not supported by linux kernels prior to v4.14. There are older kernels (like v4.4) that are still actively supported (till January 2027). 
* Should the macro guard for this section instead be something like:
```
#if defined(OPENSSL_LINUX) && defined(MADV_WIPEONFORK)
```

Reply:
It's the existing behaviour and it has worked flawlessly.
It's just to increase support for the feature that we attempt to use a hard-coded value in case the define doesn't exist.
```
madvise(addr, (size_t)page_size, MADV_WIPEONFORK)
```
will fail if the feature is not supported.